cli/src/disassemble.rs

```diff
@@ -0,0 +1,173 @@
+        $read.read_exact(&mut bytes)?;
+        <$tpe>::from_be_bytes(bytes)
+    }};
+}
+
+/// counts the number of digits of an u64 in base ten
+fn digits(val: u64) -> usize {
+    successors(Some(val), |n| (*n >= 10).then(|| n / 10)).count()
+}
+
+fn load_constants(reader: &mut impl Read) -> io::Result<Vec<String>> {
+    let mut constants = Vec::new();
+    let pool_length = read!(reader, u32);
+
+    for _ in 0..pool_length {
+        let str_len = read!(reader, u64) as usize;
+
+        let mut buff = Vec::new();
+        buff.resize(str_len, 0);
+        reader.read_exact(&mut buff)?;
+
+        let str = String::from_utf8(buff).expect("read String is not utf8");
+        constants.push(str);
+    }
+
+    Ok(constants)
+}
+
+fn display_constants(constants: &[String]) {
+    println!("Constant Pool: ");
+    let idx_pan = digits(constants.len() as u64);
+    for (idx, str) in constants.iter().enumerate() {
+        println!("#{idx:<idx_pan$}: \"{str}\"");
+    }
+}
+
+fn display_function(cursor: &mut Cursor<&[u8]>, constants: &[String]) -> io::Result<()> {
+    let name = &constants[read!(cursor, u32) as usize];
+
+    let locals_byte_count = read!(cursor, u32);
+    let parameters_bytes_count = read!(cursor, u32);
+    let return_bytes_count = read!(cursor, u8);
+
+    let instruction_count = read!(cursor, u32);
+
+    let instruction_pad = digits(instruction_count as u64);
+
+    println!("{name}:");
+    println!("\tlocals      : {locals_byte_count} bytes (including {parameters_bytes_count} bytes used for parameters)");
+    println!("\treturn      : {return_bytes_count} bytes");
+    println!("\tinstructions: {instruction_count} bytes");
+
+    let start_pos = cursor.position();
+    let end_pos = start_pos + instruction_count as u64;
+
+    while cursor.position() < end_pos {
+        let instruction = cursor.position() - start_pos;
+        let opcode = Opcode::try_from(read!(cursor, u8)).expect("Unknown opcode");
+        let mnemonic = get_opcode_mnemonic(opcode);
+        print!("\t\t#{instruction:<instruction_pad$}: {mnemonic:7} ");
+
+        match opcode {
+            Opcode::PushInt => print!("<value {}>", read!(cursor, i64)),
+            Opcode::PushByte => print!("<value {}>", read!(cursor, u8)),
+            Opcode::PushFloat => print!("<value {}>", read!(cursor, f64)),
+            Opcode::PushString => {
+                let constant_idx = read!(cursor, u32) as usize;
+                let str = &constants[constant_idx];
+                let padding = (instruction_pad - digits(constant_idx as u64)) + 10;
+                print!("<constant #{constant_idx}> {:padding$} // \"{str}\"", "")
+            }
+            Opcode::Invoke => {
+                let constant_idx = read!(cursor, u32) as usize;
+                let str = &constants[constant_idx];
+                let padding = (instruction_pad - digits(constant_idx as u64)) + 10;
```

Comment:
```suggestion
                let padding = (digits(constants.len() as u64) - digits(constant_idx as u64)) + 10;
```
See the previous explanation.